File: math/permutation-sequence/permutation-sequence.go
func getPermutation(n int, k int) string {
	nums := []int{}
	str := []byte{}

	fact := 1

	for i := 1; i <= n; i++ {
		fact *= i
		nums = append(nums, i)
	}

	k--

	for i := 0; i < n; i++ {
		fact /= (n - i)
		index := k / fact
		str = append(str, byte(nums[index]+'0'))
		nums = append(nums[:index], nums[index+1:]...)
		k -= index * fact
	}

	return string(str)
}

File: math/permutation-sequence/permutation-sequence.java
class Solution {
  public String getPermutation(int n, int k) {
    List<Integer> nums = new LinkedList<>();
    StringBuilder sb = new StringBuilder();

    int fact = 1;

    for (int i = 1; i <= n; i++) {
      fact *= i;
      nums.add(i);
    }

    k--;

    for (int i = 0; i < n; i++) {
      fact /= n - i;
      int index = k / fact;
      sb.append(nums.get(index));
      nums.remove(index);
      k -= index * fact;
    }

    return sb.toString();
  }
}
